2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala

```diff
@@ -101,93 +185,69 @@ class CollectorService(
-                      complete(
-                        responseHandler.cookie(
-                          rawRequest match {
-                            case CollectorService.QuerystringExtractor(qs) => qs
-                            case _ => ""
-                          },
-                          null,
-                          reqCookie,
-                          userAgent,
-                          host,
-                          ip,
-                          request,
-                          refererURI,
-                          "/" + path1 + "/" + path2,
-                          true
-                        )._1
-                      )
+      } ~
+      get {
+        path(Segment / Segment) { (path1, path2) =>
+          cookieIfWanted(cookieName) { reqCookie =>
+            optionalHeaderValueByName("User-Agent") { userAgent =>
+              optionalHeaderValueByName("Referer") { refererURI =>
+                headerValueByName("Raw-Request-URI") { rawRequest =>
+                  hostName { host =>
+                    clientIP { ip =>
+                      requestInstance{ request =>
+                        complete(
+                          responseHandler.cookie(
+                            rawRequest match {
+                              case CollectorService.QuerystringExtractor(qs) => qs
+                              case _ => ""
+                            },
+                            null,
+                            reqCookie,
+                            userAgent,
+                            host,
+                            ip,
+                            request,
+                            refererURI,
+                            "/" + path1 + "/" + path2,
+                            true,
+                            null
+                          )._1
+                        )
+                      }
                     }
                   }
                 }
               }
             }
           }
         }
-      }
-    } ~
-    options {
-      requestInstance { request =>
-        complete(responseHandler.preflightResponse(request))
-      }
-    } ~
-    get {
-      path("""crossdomain\.xml""".r) { path =>
-        complete(responseHandler.flashCrossDomainPolicy)
-      }
-    } ~
-    complete(responseHandler.notFound)
+      } ~
+      options {
+        requestInstance { request =>
+          complete(responseHandler.preflightResponse(request))
+        }
+      } ~
+      get {
+        path("""crossdomain\.xml""".r) { path =>
+          complete(responseHandler.flashCrossDomainPolicy)
+        }
+      } ~
+      complete(responseHandler.notFound)
+  }
+
+  /**
+    * Directive to extract a cookie if a cookie name is specified and if such a cookie exists
+    *
+    * @param name Optionally configured cookie name
+    * @return Directive1[Option[HttpCookie]]
+    */
+  def cookieIfWanted(name: Option[String]): Directive1[Option[HttpCookie]] = name match {
```

Comment:
Yeah, I'm going to get the collector running on a new dev box on Thursday to QA it further, if it works, we'll roll it out to all the dev boxes (blue-green style so there should be no downtime)